site/e2e/tests/signIn.spec.ts

```diff
@@ -0,0 +1,99 @@
+ *    that will affect all the other tests
+ */
+import { test, expect } from "@playwright/test";
+import { accessibleDropdownLabel } from "modules/dashboard/Navbar/UserDropdown/UserDropdown";
+import { Language } from "modules/dashboard/Navbar/UserDropdown/UserDropdownContent";
+import { createUser, getCurrentOrgId, setupApiCalls } from "../api";
+import * as constants from "../constants";
+import { assertNoUncaughtRenderError } from "../helpers";
+import { beforeCoderTest } from "../hooks";
+
+test.beforeEach(async ({ page }) => await beforeCoderTest(page));
+
+// One complication with this test is that you can't process the logout using
+// the same user credentials as all the other test cases. If this runs first and
+// logs the user out, every single other test will start failing, but Playwright
+// also doesn't give you the ability to have certain tests run last. Have to
+// make a separate user to guarantee more test isolation
+const credentials = {
+  username: "blah-blah-blah",
+  password: constants.password,
+} as const;
+
+test.skip("Signing in and out", async ({ page, baseURL }) => {
+  const handleSignIn = async () => {
+    await page.goto(`${baseURL}/login`, { waitUntil: "domcontentloaded" });
+
+    const emailField = page.getByRole("textbox", { name: /Email/ });
+    const passwordField = page.getByRole("textbox", { name: /Password/ });
+    const loginButton = page.getByRole("button", { name: /Sign in/i });
+
+    await emailField.fill(`${credentials.username}@coder.com`);
+    await passwordField.fill(credentials.password);
+    await loginButton.click();
+
+    await expect(page).toHaveURL(`${baseURL}/workspaces`);
+  };
+
+  const handleSignOut = async () => {
+    const dropdownName = new RegExp(accessibleDropdownLabel);
+    const dropdown = page.getByRole("button", { name: dropdownName });
+    await dropdown.click();
+
+    const signOutOption = page.getByRole("menuitem", {
+      name: Language.signOutLabel,
+
+      // Need to set to true because MUI automatically hides non-focused menu
+      // items from screen readers (and by extension, Playwright). Trying to tab
+      // through everything felt like it could get even more flaky
+      includeHidden: true,
+    });
+
+    await signOutOption.click();
+    await expect(page).toHaveTitle(/^Sign in to /);
+    await expect(page).toHaveURL(new RegExp(`^${baseURL}/login`));
+
+    /**
+     * 2024-05-06 - Adding this to assert that we can't have regressions around
+     * the log out flow after it was fixed.
+     * @see {@link https://github.com/coder/coder/issues/13130}
+     */
+    await assertNoUncaughtRenderError(page);
+  };
+
+  const setupNewUser = async () => {
+    const orgId = await getCurrentOrgId();
+    await createUser(orgId, credentials.username, credentials.password);
+  };
+
+  await setupApiCalls(page);
+  await setupNewUser();
+
+  await handleSignIn();
+  await handleSignOut();
```

Comment:
Why the sign-in and sing out tests are encapsulated into functions? I would expect each one to be a separated test.